### ISSUE TITLE
Marketplace update breaks builds

### ISSUE THREAD

**Issue by jeanfrancoislarente** [4tecture/BuildVersioning#17]
An update happened today and all of a sudden $(Build.BuildVersion) no longer resolves.

Doesn't look like the version was bumped but the extension is no longer working (0.9.0 I believe)

**Comment by jeanfrancoislarente** [comment]
In case it helps anyone else...

I was using BuildVersion.Version for the longest time and since ~10AM today I need to use

<ReferenceVariable>.BuildVersion.Version

ReferenceVariable is set in the Output Variables section

**Comment by marc-mueller** [comment]
Hi @jeanfrancoislarente 

Thank you for your message. Could you please provide more details? Where did you get the variable `Build.BuildVersion` from? The task creates the variable `BuildVersion.Version` and this seems to work as expected. 

Marc

This is a log output of a 0.9.0 test pipeline, you'll see all the set variables with GitVersion version provider:
```

##[debug]set BuildVersion.PreReleaseTag=.27
##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.PreReleaseTag;issecret=false;].27
##[debug]BuildVersion.PreReleaseTag=.27
##[debug]Setting variable BuildVersion.PreReleaseTag to value .27
##[debug]set BuildVersion.PreReleaseTagWithDash=
##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.PreReleaseTagWithDash;issecret=false;]
##[debug]BuildVersion.PreReleaseTagWithDash=
##[debug]Setting variable BuildVersion.PreReleaseTagWithDash to value 
##[debug]set BuildVersion.PreReleaseLabel=
##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.PreReleaseLabel;issecret=false;]
##[debug]BuildVersion.PreReleaseLabel=
##[debug]Setting variable BuildVersion.PreReleaseLabel to value 
##[debug]set BuildVersion.PreReleaseNumber=
##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.PreReleaseNumber;issecret=false;]
##[debug]BuildVersion.PreReleaseNumber=
##[debug]Setting variable BuildVersion.PreReleaseNumber to value 
##[debug]set BuildVersion.WeightedPreReleaseNumber=
##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.WeightedPreReleaseNumber;issecret=false;]
##[debug]BuildVersion.WeightedPreReleaseNumber=
##[debug]Setting variable BuildVersion.WeightedPreReleaseNumber to value 
##[debug]set BuildVersion.BuildMetaData=
##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.BuildMetaData;issecret=false;]
##[debug]BuildVersion.BuildMetaData=
##[debug]Setting variable BuildVersion.BuildMetaData to value 
##[debug]set BuildVersion.BuildMetaDataPadded=
##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.BuildMetaDataPadded;issecret=false;]
##[debug]BuildVersion.BuildMetaDataPadded=
##[debug]Setting variable BuildVersion.BuildMetaDataPadded to value 
##[debug]set BuildVersion.FullBuildMetaData=Branch.master.Sha.90af50d84155d359264653fb05411c996ca80d15
##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.FullBuildMetaData;issecret=false;]Branch.master.Sha.90af50d84155d359264653fb05411c996ca80d15
##[debug]BuildVersion.FullBuildMetaData=Branch.master.Sha.90af50d84155d359264653fb05411c996ca80d15
##[debug]Setting variable BuildVersion.FullBuildMetaData to value Branch.master.Sha.90af50d84155d359264653fb05411c996ca80d15
##[debug]set BuildVersion.Sha=90af50d84155d359264653fb05411c996ca80d15
##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.Sha;issecret=false;]90af50d84155d359264653fb05411c996ca80d15
##[debug]BuildVersion.Sha=90af50d84155d359264653fb05411c996ca80d15
##[debug]Setting variable BuildVersion.Sha to value 90af50d84155d359264653fb05411c996ca80d15
##[debug]set BuildVersion.ShortSha=90af50d
##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.ShortSha;issecret=false;]90af50d

##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.BuildCounter;issecret=false;]0
##[debug]Setting variable BuildVersion.BuildCounter to value 0
##[debug]paramVersionCode=2
##[debug]set BuildVersion.Version=0.1.27.0
##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.Version;issecret=false;]0.1.27.0
##[debug]BuildVersion.Version=0.1.27.0
##[debug]Setting variable BuildVersion.Version to value 0.1.27.0
##[debug]set BuildVersion.VersionInfo=0.1.27
##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.VersionInfo;issecret=false;]0.1.27
##[debug]BuildVersion.VersionInfo=0.1.27
##[debug]Setting variable BuildVersion.VersionInfo to value 0.1.27
##[debug]set BuildVersion.SemVer=0.1.27
##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.SemVer;issecret=false;]0.1.27
##[debug]BuildVersion.SemVer=0.1.27
##[debug]Setting variable BuildVersion.SemVer to value 0.1.27
##[debug]set BuildVersion.AssemblySemVer=0.1.27.0
##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.AssemblySemVer;issecret=false;]0.1.27.0
##[debug]BuildVersion.AssemblySemVer=0.1.27.0
##[debug]Setting variable BuildVersion.AssemblySemVer to value 0.1.27.0
##[debug]set BuildVersion.VersionCode=1051
##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.VersionCode;issecret=false;]1051
##[debug]BuildVersion.VersionCode=1051
##[debug]Setting variable BuildVersion.VersionCode to value 1051
##[debug]The following versions will be used within this task's actions:
##[debug]AssemblyVersion:                0.1.27.0
##[debug]Assembly File Version:          0.1.27.0
##[debug]Assembly Informational Version: 0.1.27
##[debug]Update build number for build: 176 to: 0.1.27 at backend.
##[debug]Processed: ##vso[build.updatebuildnumber]0.1.27
##[debug]set BuildVersion.BuildNumber=0.1.27
##[debug]Processed: ##vso[task.setvariable variable=BuildVersion.BuildNumber;issecret=false;]0.1.27
##[debug]doReplaceAssemblyInfo=false
##[debug]doReplaceNuspec=false
##[debug]doReplaceDotNetCore=true
##[debug]filePatternDotNetCore=**\*.csproj
##[debug]sourcePath=/home/vsts/work/1/s
##[debug]check path : /home/vsts/work/1/s
##[debug]doShowWarningsForUnmatchedRegex=true
##[debug]paramDotNetCoreVersionType=3
##[debug]Using the following versions for project files:
##[debug]Assembly Version:             0.1.27.0
##[debug]Assembly File Version:        0.1.27.0
##[debug]Package Version:              0.1.27
```

**Comment by SpectrumDynamics** [comment]
same issue. after update 0.9 not resolved: BuildVersion.MajorVersion, BuildVersion.MinorVersion,BuildVersion.PatchVersion,BuildVersion.BuildCounter.
I use variable as version source. variable defined as 1.2.3.0.
Build Counter working well and values are resolved, but BuildVersion.* not available in the next task.
Build Counter key:
$(Build.Repository.Name)-$(BuildVersion.MajorVersion).$(BuildVersion.MinorVersion).$(BuildVersion.PatchVersion) - change build name to <REPO>_1.2.3.45

**Comment by marc-mueller** [comment]
We are working on a fix, hopefully within the next hour....

**Comment by marc-mueller** [comment]
We apologize for the error. Unfortunately, there was a bug with newly introduced output variables and our validation build did verify other variables. The variable behavior is now the same as before but with the newly added additional gitversion variables.

We published version 0.9.1 which fixes the issue.

**Comment by jeanfrancoislarente** [comment]
Fix confirmed, thanks!
